src/frontend/type_checker.cpp

```diff
@@ -269,9 +293,17 @@ void TypeChecker::visit(FuncDecl::Ptr decl) {
       reportMultipleDefs("function parameter", name, genericParam);
     }
 
-    const Type::Ptr type = (genericParam->type == GenericParam::Type::RANGE) ?
-                           to<Type>(makeTensorType(ScalarType::Type::INT)) :
-                           to<Type>(env.getSetDefinition(genericParam));
+    Type::Ptr type;
+    if (genericParam->type == GenericParam::Type::RANGE) {
+      type = to<Type>(makeTensorType(ScalarType::Type::INT));
+    }
+    else {
+      SetType::Ptr setDef = env.getSetDefinition(genericParam);
+      // We have to guard for undefined nullptr set definitions.
+      if (setDef) {
```

Comment:
I don't think it should actually be possible for `setDef` to be null here unless there's a bug somewhere else in the type checker (presumably with `ComputeSetDefinitions`, which is responsible for setting the `setDef`'s).